src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -369,6 +380,242 @@ void CascadiaSettings::_LoadDynamicProfiles()
+{
+    // First, accumulate the namespaces the user wants to ignore
+    std::unordered_set<std::wstring> ignoredNamespaces;
+    const auto disabledProfileSources = CascadiaSettings::_GetDisabledProfileSourcesJsonObject(_userSettings);
+    if (disabledProfileSources.isArray())
+    {
+        for (const auto& json : disabledProfileSources)
+        {
+            ignoredNamespaces.emplace(JsonUtils::GetValue<std::wstring>(json));
+        }
+    }
+
+    // Search through the local app data folder
+    if (std::filesystem::exists(wil::ExpandEnvironmentStringsW<std::wstring>(LocalAppDataFolder.data())))
+    {
+        _ApplyJsonStubsHelper(LocalAppDataFolder, ignoredNamespaces);
+    }
+
+    // Search through the global app data folder
+    if (std::filesystem::exists(wil::ExpandEnvironmentStringsW<std::wstring>(ProgramDataFolder.data())))
+    {
+        _ApplyJsonStubsHelper(ProgramDataFolder, ignoredNamespaces);
+    }
+
+    // Search through app extensions
+    // Gets the catalog of extensions with the name "com.microsoft.windows.terminal.settings"
+    const auto catalog = Windows::ApplicationModel::AppExtensions::AppExtensionCatalog::Open(winrt::to_hstring(AppExtensionHostName));
+
+    // Extracting the list of extensions from the catalog is an async operation, but since
+    // we are on a UI thread - and so cannot call a blocking operation like .get() - we use
+    // a mutex and condition variable
+    std::condition_variable cv;
+    std::mutex mtx;
+    Windows::Foundation::Collections::IVectorView<Windows::ApplicationModel::AppExtensions::AppExtension> extensions;
+
+    auto lambda = [&]() -> winrt::fire_and_forget {
+        co_await resume_background();
+        const auto localExtensions = catalog.FindAllAsync().get();
+
+        std::unique_lock<std::mutex> lock{ mtx };
+        extensions = localExtensions;
+        cv.notify_all();
+    };
+
+    std::unique_lock<std::mutex> lock{ mtx };
+    lambda();
+    cv.wait(lock);
+
+    for (auto ext : extensions)
+    {
+        // Only apply the stubs if the package name is not in ignored namespaces
+        if (ignoredNamespaces.find(ext.Package().DisplayName().c_str()) == ignoredNamespaces.end())
+        {
+            // Likewise, getting the public folder from an extension is an async operation
+            // So we use another mutex and condition variable
+            std::condition_variable cv2;
+            std::mutex mtx2;
+            Windows::Storage::StorageFolder foundFolder{ nullptr };
+
+            auto lambda2 = [&]() -> winrt::fire_and_forget {
+                co_await resume_background();
+                const auto localFolder = ext.GetPublicFolderAsync().get();
+                std::unique_lock<std::mutex> lock{ mtx2 };
+                foundFolder = localFolder;
+                cv2.notify_all();
+            };
+
+            std::unique_lock<std::mutex> lock2{ mtx2 };
+            lambda2();
+            cv2.wait(lock2);

```

Comment:
Same here regarding `wil::wait_for_completion`

Reply:
@DHowett and I mused about this for half an hour and everything else is so much worse. Forget it.